src/__tests__/apply.test.js

```diff
@@ -0,0 +1,22 @@
+/* eslint-env node, jest */
+const { apply } = require('..')
+
+describe('apply', function () {
+  let testFn = jest.fn()
+
+  beforeEach(function () {
+    jest.resetAllMocks()
+  })
+
+  it('passes the function the iterable of arguments provided to it', function () {
+    apply(testFn, [2, 3])
+    expect(testFn).toHaveBeenCalledTimes(1)
+    expect(testFn).toHaveBeenLastCalledWith(2, 3)
```

Comment:
I always prefer _one assertion line per test unit_ (i.e. 1 `expect` per `it`). In this case:

```typescript
apply(testFn, [2, 3])
expect(testFn.mock.calls).toEqual([
  [2, 3]
])
```

Reply:
I don't care. I just simply do not begin to care. I'm sorry. You want to control the precise details of the code style, you write the feature.

Reply:
Dude, calm down. I do not know if you had a bad day, but "I don't care" is not really a resonable answer.

Whilst [my other suggestion](https://github.com/sithmel/iter-tools/pull/205#discussion_r261835609) might be mere code style, this one acts differently on failed tests.

Reply:
My point was that it isn't a user facing distinction. It doesn't affect correctness or performance and it doesn't affect test correctness or performance. Since we often have rather intense discussion of how things should be designed or the structure of implementation, I try my best to avoid further nitpicking. We're just doing this because we enjoy doing it, and at the end of the day everyone will have slightly different ideas and patterns as long as we keep the differences away from the user that's OK.

Getting into the specifics of your suggestion, I don't find it to be particularly compelling because it is not internally consistent with other best practices. For example a competing best practice for the understandability of test debugging output is to use the most relevant possible matchers so that the debugging output is able to convey exactly what the expectation is. For example `toHaveBeenCalled` has better output than `toEqual`. This is a pattern that is broadly accepted across a variety of frameworks, all of which structure themselves to allow you to make multiple assertions. Basically if I wanted to wholeheartedly embrace the pattern you are suggesting everywhere, it would imply a different test framework than the one we are using.

Secondly I already took the practical step to reduce the risk of debugging confusion here, where is to put the more basic assumption first. If I make the call and the function isn't called at all, I don't see the point of proceeding further into the test like trying output the difference in arguments. That won't be why the test failed, it will have failed because the function wasn't called. So I would assert that my version already aligns with the intent of the framework we are using, and results in the most readable debugging output.

I'm sorry I didn't feel like writing this all out earlier.

Reply:
I guess you have your point. But I still insist on [this](https://github.com/sithmel/iter-tools/pull/205#discussion_r261835886)